rundeckapp/grails-app/services/rundeck/services/ScheduledExecutionService.groovy

```diff
@@ -2937,9 +2937,10 @@ class ScheduledExecutionService implements ApplicationContextAware, Initializing
             }.findAll{it}
         }
         def addedNotifications=[]
+        if(!scheduledExecution.notifications) replaceAll = true
```

Comment:
(This was one of the first changes in the branch)
Since we now have multiple notifications of the same type, I'm not sure if we should always replace all notifications, we don't have something like notificationID to update the corresponding notifications only.